### ISSUE TITLE
[feature] let users request to be added to a project

### ISSUE THREAD

**Issue by ctb** [CenterForOpenScience/osf.io#2288]
It would be nice if users could go to a public project and be asked to be added to it - e.g. when running workshops that are using the collaborative wiki, this would let the workshop coordinator approve requests quickly without having to search for all the accounts, etc.  Same for classes and courses over a longer time period.

For example,at

https://osf.io/26aun/

it would be nice to have a button next to "Public" on the upper right that let me request membership.

**Comment by lbanner** [comment]
Thank you @ctb .... so, different from the "watch" option, yes? In your scenario, is the workshop coordinator owner of the public project? And you, as an OSF user, wish to add to and/or receive notification about the project?

**Comment by ctb** [comment]
Yes, different from the watch option.  The workshop coordinator would be an administrator (capable of approving membership in the public project), and I, as an OSF user, wish to be able to contribute to the project (upload files, edit wiki).

**Comment by lbanner** [comment]
Good -- thanks again, Titus!

**Comment by brianjgeiger** [comment]
Thanks, @ctb. We are working on some modifications to the sharing controls which would support this. In the last epic, we got notifications in, so we could email project administrators when requests like that would happen. We have a proposal from one of our developers #2232 to allow users who are contributors to get higher permissions, which would add the administrative side of these kinds of requests to be handled. After that, your suggestion would fit in nicely.

My only concern, and you'll see some of that addressed in #2232, is that we want to avoid having Project Admins spammed by people who aren't going to be added as contributors. You might want to check that proposal and see if that kind of solution would make sense for your needs.

**Comment by ctb** [comment]
The proposed scheme looks good to me!
